include/cucumber-cpp/internal/utils/qtCapture.hpp

```diff
@@ -0,0 +1,182 @@
+/* Based on Sir Digby Chicken post from:
+ * http://stackoverflow.com/questions/5419356/redirect-stdout-stderr-to-a-string
+ * who granted his work on BSD license
+ * adjusted for Qt by konserw
+ */
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) && !defined(__CYGWIN__)
+#define WINDOWS
+#include <io.h>
+#include <fcntl.h>
+#define popen _popen 
+#define pclose _pclose
+#define stat _stat 
+#define dup _dup
+#define dup2 _dup2
+#define fileno _fileno
+#define close _close
+#define pipe _pipe
+#define read _read
+#define eof _eof
+#else
+#include <errno.h>
+#endif
+
+#include <cstdio>
+#include <QtConcurrent/QtConcurrent>
+
+#ifndef STD_OUT_FD
+#define STD_OUT_FD (fileno(stdout))
+#endif
+#ifndef STD_ERR_FD
+#define STD_ERR_FD (fileno(stderr))
+#endif
+
+class qtCapture {
```

Comment:
None of this code is dependent on a template parameter, so it can all be moved out of this header and into a .cpp file.